tensorflow/core/kernels/conv_ops.cc

```diff
@@ -868,8 +872,18 @@ void LaunchConv2DOp<GPUDevice, T>::operator()(
   const bool compute_in_nhwc = DataTypeToEnum<T>::value == DT_HALF &&
                                stream->GetCudaComputeCapability().IsAtLeast(
                                    se::CudaComputeCapability::VOLTA);
+#elif TENSORFLOW_USE_ROCM && TF_ROCM_VERSION >= 50000
+  // AMD Matrix Cores on MI100 and MI200 allow for efficient FP16 NHWC convolutions
```

Comment:
perhaps create an explicit routine `bool useNhwcLayoutForConvOnRocm()` here...see similar comment for batch_norm below